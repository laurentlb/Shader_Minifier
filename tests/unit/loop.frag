float f(inout float noinline){return 0.;}

float a;
void main()
{
	float c;
	for (float d=0.; d<50.; ++d)
	{
		c+=cos(d);
	}
	float b;
	for (b=0.; b<50.; ++b)
	{
		c+=cos(c);
	}
	b = a;
	while (b<50.)
	{
		c+=cos(c);
		b++;
	}
	b = a;
	while (b<50.)
	{
		b++;
	}
	b = a;
	while (b<50.)
	{
		c+=cos(c);
		float arg; float d=f(arg); // d prevents moving b+=d to a for
		b+=d;
	}
	b = a;
	while (b<50.)
	{
		if (a < b) continue; // continue prevents moving b++ to a for
		b++;
	}
	b = a;
	while (b<50.)
	{
		if (a < b) a=b; else continue; // continue prevents moving b++ to a for
		b++;
	}
	for (; b<50.; ++b)
	{
		c+=cos(b);
	}
	do
		b++;
	while (b<100.);
	do {
		if (a>10.)b++;
		b+=a;
	} while (b<100.);
}
