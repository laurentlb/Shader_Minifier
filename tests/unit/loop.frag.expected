/* File generated with 
 * http://www.ctrl-alt-test.fr
 */
#ifndef LOOP_FRAG_EXPECTED_
# define LOOP_FRAG_EXPECTED_

const char *loop_frag =
 "float f()"
 "{"
   "return 0.;"
 "}"
 "void main()"
 "{"
   "float c,b;"
   "for(float a=0;a<50.;++a)"
     "c+=cos(a);"
   "for(b=0;b<50.;++b)"
     "c+=cos(c);"
   "for(b=a;b<50.;b++)"
     "c+=cos(c);"
   "for(b=a;b<50.;b++)"
     ";"
   "for(b=a;b<50.;)"
     "{"
       "c+=cos(c);"
       "float d=f();"
       "b+=d;"
     "}"
   "for(b=a;b<50.;)"
     "{"
       "if(a<b)"
         "continue;"
       "b++;"
     "}"
   "for(b=a;b<50.;)"
     "{"
       "if(a<b)"
         "a=b;"
       "else"
         " continue;"
       "b++;"
     "}"
   "for(;b<50.;++b)"
     "c+=cos(b);"
 "}";

#endif // LOOP_FRAG_EXPECTED_
